### PR TITLE
feat: migrate to viem

### PR DESCRIPTION
**Pull request by nemmtor**

### CODE REVIEW

File: src/application/polymarket/types.ts [hunk 1]
```diff
@@ -80,31 +76,83 @@ export interface PolymarketUser {
...
+  orderType: 'FOK';
+}
+
+export interface OrderData {
+  maker: string;
+  taker: string;
+  tokenId: string;
+  makerAmount: string;
+  takerAmount: string;
+  side: 0;
```
Comment (SolutionsEngineer) [inline src/application/polymarket/types.ts:131]: Are the values like 'BUY', 'FOK', '0', and '2' static, non-changing values?

File: src/application/polymarket/utils.ts [hunk 2]
```diff
@@ -35,3 +61,275 @@ export const isEventUrl = (url: string) => {
...
+        type: 'uint256',
+      },
+    ],
+  } as const;
+
+  const message = {
+    address: address,
+    timestamp,
+    nonce: BigInt(0),
+    message: 'This message attests that I control the given wallet',
```
Comment (SolutionsEngineer) [inline src/application/polymarket/utils.ts:103]: Just to make sure, was that message supposed to be a value or a comment?

File: src/application/polymarket/widgets/market-widget.tsx [hunk 3]
```diff
@@ -209,8 +269,9 @@ export const Market = ({
                               <InputBase.Label label="Amount" />
                               <Chip className="bg-[#2C3F4F]">
                                 Balance $
-                                {user.safeWalletDetails?.balance.toFixed(2) ??
-                                  0}
+                                {(user.safeWalletDetails?.balance ?? 0).toFixed(
+                                  2,
+                                ) ?? 0}
```
Comment (SolutionsEngineer) [inline src/application/polymarket/widgets/market-widget.tsx:274]: I think we can remove this fallback as we have one in the toFixed source

File: src/shared/web3/types.ts [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
-  nativeCurrency: {
-    name: string;
-    symbol: string;
-    decimals: number;
-  };
-  rpcUrls: string[];
-  blockExplorerUrls: string[];
-}
-
+// export interface Chain {
```
Comment (SolutionsEngineer) [inline src/shared/web3/types.ts:6]: Do we want to keep that interface commented?

**Review by SolutionsEngineer — APPROVED** [review]
I've left a few minor comments, but overall, it looks great 👌

File: src/application/polymarket/types.ts (hunk 1 above)
Comment (nemmtor) [reply src/application/polymarket/types.ts:131]: it always has to be like that for our use-case, in general polymarket accepts some different values but we only care about these when making an order

File: src/application/polymarket/utils.ts (hunk 2 above)
Comment (nemmtor) [reply src/application/polymarket/utils.ts:103]: it was supposed to be a value, it is the one visible in metamask when you are asked to sign

File: src/application/polymarket/widgets/market-widget.tsx (hunk 3 above)
Comment (nemmtor) [reply src/application/polymarket/widgets/market-widget.tsx:274]: good catch!

File: src/shared/web3/types.ts (hunk 4 above)
Comment (nemmtor) [reply src/shared/web3/types.ts:6]: hell no 😅

File: src/application/polymarket/types.ts (hunk 1 above)
Comment (SolutionsEngineer) [reply src/application/polymarket/types.ts:131]: okay

**Review by SolutionsEngineer — DISMISSED** [review]

**Pull request merged**